### ISSUE TITLE
Unexpected ERR_UseDefViolation error reported within an 'if' block

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#9121]
```
public class X
{
    public static void Main()
    {
    }

    void Dummy<T>(T x) {}

    void Test6()
    {
        if (x6 && true is var x6)
            Dummy(x6);
    }
}
```

Observed:

```
    // (12,13): error CS0841: Cannot use local variable 'x6' before it is declared
    //         if (x6 && true is var x6)
    Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "x6").WithArguments("x6").WithLocation(12, 13),
    // (13,19): error CS0165: Use of unassigned local variable 'x6'
    //             Dummy(x6);
    Diagnostic(ErrorCode.ERR_UseDefViolation, "x6").WithArguments("x6").WithLocation(13, 19)
```

Expected:
No ERR_UseDefViolation error reported

**Comment by gafter** [comment]
Fixed in https://github.com/dotnet/roslyn/issues/25774
